### DISCUSSION TITLE
How to fetch tag message in github actions

### DISCUSSION THREAD

**Discussion by nitin-jci** [community/community#27155] [Actions]
<p>I want to capture the github tag message in Github Actions.</p>
<p>If user pushes tag with some message e.g.</p>
<p><code>git tag -a 1.0.0 -m "Test Message"</code></p>
<p>than, I would like to capture that message(“Test Message”) inside the github actions.</p>
<p>Any help or reference is much appreciated.</p>

**Comment by airtower-luna** [comment]
<p>For a one-line message you can use:</p>
<pre><code class="lang-bash">git tag -l --format='%(contents:subject)' 1.0.0
</code></pre>
<p>With a multiline message this will give you just the title line. For more formatting options check the formatting options described here:</p><aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://git-scm.com/favicon.ico" class="site-icon" width="16" height="16">
      <a href="https://git-scm.com/docs/git-for-each-ref" target="_blank" rel="noopener">git-scm.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="16" height="16">

<h3><a href="https://git-scm.com/docs/git-for-each-ref" target="_blank" rel="noopener">Git - git-for-each-ref Documentation</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by nitin-jci** [comment]
<p>Thank you for the reply.<br>
I found the issue. It was related to the git version which we were using on self-hosted runner. When I verified the CHECKOUT log, I found that because of the older version, github actions was downloading the repo using Rest API instead of cloning it.</p>
<pre><code class="lang-auto">  git version 2.17.1
Deleting the contents of &lt;YOUR REPO&gt;
**The repository will be downloaded using the GitHub REST API**
To create a local Git repository instead, add Git 2.18 or higher to the PATH
Downloading the archive
Writing archive to disk
Extracting the archive
</code></pre>
<p>Therefore, my actions was not able to the run the git command. I have upgraded the version and it fixed the issue.</p>

**Comment by nitin-jci** [comment]
<p>Sorry, I commented too early that solution worked. Following command works fine on my local. However, on build machine it doesn’t:</p>
<pre><code class="lang-auto">git tag -l --format='%(contents:subject)' 1.0.0
</code></pre>
<p>it returns git commit message instead of git tag message “Test Message”.</p>

**Comment by airtower-luna** [comment]
<p>That’s odd… I wonder if the behavior is different between Git versions, intentionally or due to a bug?</p>

**Comment by nitin-jci** [comment, marked as answer]
<p>So, I tried to increasing the fetch-depth as well by setting it up as 0 at checkout step. But that didn’t help even. However, I have used the following command before the command that you helped me with and it worked. :slight_smile:</p>
<p>git fetch --depth=1 origin +refs/tags/<em>:refs/tags/</em></p>
<p>So, final solution is :</p>
<pre><code class="lang-auto">git fetch --depth=1 origin +refs/tags/*:refs/tags/*
git tag -l --format='%(contents:subject)' 1.0.0
</code></pre>

**Reply by hedon954** [reply]
Thanks for your solution! It helps me lots.
